src/credentials/CertificationDeclaration.cpp

```diff
@@ -245,57 +283,74 @@ CHIP_ERROR DecodeCertificationElements(const ByteSpan & encodedCertElements, Cer
 
 bool CertificationElementsDecoder::IsProductIdIn(const ByteSpan & encodedCertElements, uint16_t productId)
 {
-    VerifyOrReturnError(PrepareToReadProductIdList(encodedCertElements) == CHIP_NO_ERROR, false);
+    VerifyOrReturnError(FindAndEnterArray(encodedCertElements, ContextTag(kTag_ProductIdArray)) == CHIP_NO_ERROR, false);
 
     uint16_t cdProductId = 0;
-    CHIP_ERROR error     = CHIP_NO_ERROR;
-
-    while ((error = GetNextProductId(cdProductId)) == CHIP_NO_ERROR)
+    while (GetNextProductId(cdProductId) == CHIP_NO_ERROR)
     {
         if (productId == cdProductId)
         {
             return true;
         }
     }
-
     return false;
 }
 
-CHIP_ERROR CertificationElementsDecoder::PrepareToReadProductIdList(const ByteSpan & encodedCertElements)
+CHIP_ERROR CertificationElementsDecoder::FindAndEnterArray(const ByteSpan & encodedCertElements, Tag arrayTag)
 {
-    mIsInitialized                = false;
-    mCertificationDeclarationData = encodedCertElements;
+    TLVType outerContainerType1;
+    TLVType outerContainerType2;
 
-    mReader.Init(mCertificationDeclarationData);
+    mReader.Init(encodedCertElements);
     ReturnErrorOnFailure(mReader.Next(kTLVType_Structure, AnonymousTag()));
-    ReturnErrorOnFailure(mReader.EnterContainer(mOuterContainerType1));
+    ReturnErrorOnFailure(mReader.EnterContainer(outerContainerType1));
 
     // position to ProductId Array
```

Comment:
This comment does not match what this function does now.

Reply:
updated